python/pyspark/sql/functions.py

```diff
@@ -1862,9 +1862,33 @@ def udf(f, returnType=StringType()):
     >>> slen = udf(lambda s: len(s), IntegerType())
     >>> df.select(slen(df.name).alias('slen')).collect()
     [Row(slen=5), Row(slen=3)]
+    >>> @udf
```

Comment:
I like this example as a doctest - but maybe good to also show these being applied in the doctest.